### PR TITLE
docs(flex): expands design guidelines for flex layout.

### LINKED ISSUE #3581: Enhance design guidelines for our Layouts pages - Flex
Add guidelines to Flex Layouts page
Include picture example of Flex being used

### PR DESCRIPTION
**Pull request by edonehoo**
closes #3581

### CODE REVIEW

File: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md [hunk 1]
```diff
@@ -2,4 +2,36 @@
 id: Flex
 section: layouts
 ---
-The **Flex** layout is a tool to build your own custom layout that builds-in the PatternFly spacer and breakpoint systems. This enables developers to create a custom layout that can have consistent spacing and change responsively with other PatternFly layouts and components.
+
+The **flex** layout supports a completely custom layout by utilizing the PatternFly spacer and breakpoint systems. Flex layouts are infinitely nestable and allow you to adjust spacing, direction, alignment, justification, wrapping, and width to fit your needs.
+
+## Elements
+
+As an example, our [dashboard pattern](/patterns/dashboard) utilizes a flex layout.
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md:10]:
```suggestion
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md [hunk 2]
```diff
@@ -2,4 +2,36 @@
...
+
+For example, our [dashboard pattern](/patterns/dashboard) combines the grid layout with flex to improve the responsiveness of information. Similarly, our [primary-detail pattern](patterns/primary-detail/react-demos/primary-detail-full-page/) combines the stack layout with flex to ensure that icons and buttons are responsive to changes in screen size. 
+
+### When to use 
+
+Use the flex layout with content heavy UIs and when it is necessary to wrap UI elements.
+
+### When to use the flex layout vs the grid layout
+
+Use the flex layout when a rigid grid is not necessary or wanted. If you need a more structured layout, consider using a grid.
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md:29]:
```suggestion
Use the flex layout when a rigid grid is not necessary or wanted. If you need a more structured layout, consider using a the [grid layout](/layouts/grid).
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md [hunk 3]
```diff
@@ -2,4 +2,36 @@
...
+
+## Variations
+
+The flex layout is based on the [CSS "flex" properties](https://www.w3schools.com/cssref/css3_pr_flex.php), which help determine how a flex item will grow or shrink to fit the space available in its container. These properties allow for variation in the implementation and appearance of flex layouts.
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md:33]:
```suggestion
The flex layout is based on the [CSS "flex" properties](https://www.w3schools.com/cssref/css3_pr_flex.php), which help determine how a flex item will grow or shrink to fit a containers available space. These properties allow for variation in the implementation and appearance of flex layouts. 
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md [hunk 4]
```diff
@@ -2,4 +2,36 @@
...
+
+### When to use the flex layout vs the grid layout
+
+Use the flex layout when a rigid grid is not necessary or wanted. If you need a more structured layout, consider using a grid.
+
+## Variations
+
+The flex layout is based on the [CSS "flex" properties](https://www.w3schools.com/cssref/css3_pr_flex.php), which help determine how a flex item will grow or shrink to fit the space available in its container. These properties allow for variation in the implementation and appearance of flex layouts. 
+
+Most of the flex layout variations are available at our standard breakpoints. In the future, we plan to apply these modifiers via [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver).
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md:35]:
```suggestion
Most of the flex layout variations are available at our standard [breakpoints](/developer-resources/global-css-variables#breakpoint-variables-and-class-suffixes). In the future, we plan to apply these modifiers via [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver). 
```

**Comment by edonehoo** [comment]
@lboehling @mmenestr

File: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md [hunk 5]
```diff
@@ -2,4 +2,35 @@
...
+
+### When to use the flex layout vs the grid layout
+
+Use the flex layout when a rigid grid is not necessary or wanted. If you need a more structured layout, consider using a the [grid layout](/layouts/grid).
+
+## Variations
+
+The flex layout is based on the [CSS "flex" properties](https://www.w3schools.com/cssref/css3_pr_flex.php), which help determine how a flex item will grow or shrink to fit a containers available space. These properties allow for variation in the implementation and appearance of flex layouts. 
+
+Most of the flex layout variations are available at our standard [breakpoints](/developer-resources/global-css-variables#breakpoint-variables-and-class-suffixes). In the future, we plan to apply these modifiers via [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver).
```
Comment (mmenestr) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md:34]: for some reason the developer-resources link doesn't work

File: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md [hunk 6]
```diff
@@ -2,4 +2,35 @@
...
+1. **Flex:** The flex container, which contains all flex items.
+1. **Flex item:** An individual section of content to be placed in the flex container. 
+
+## Usage
+
+Flex layouts can be combined with other layouts to create versatile UIs. 
+
+For example, our [dashboard pattern](/patterns/dashboard) combines the grid layout with flex to improve the responsiveness of information. Similarly, our [primary-detail pattern](/patterns/primary-detail/react-demos/primary-detail-full-page/) combines the stack layout with flex to ensure that icons and buttons are responsive to changes in screen size. 
+
+### When to use
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md:22]:
```suggestion
### When to use the flex layout
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md [hunk 7]
```diff
@@ -2,4 +2,35 @@
...
+
+### When to use the flex layout vs the grid layout
+
+Use the flex layout when a rigid grid is not necessary or wanted. If you need a more structured layout, consider using a the [grid layout](/layouts/grid).
+
+## Variations
+
+The flex layout is based on the [CSS "flex" properties](https://www.w3schools.com/cssref/css3_pr_flex.php), which help determine how a flex item will grow or shrink to fit a containers available space. These properties allow for variation in the implementation and appearance of flex layouts. 
+
+Most of the flex layout variations are available at our standard [breakpoints](/developer-resources/global-css-variables#breakpoint-variables-and-class-suffixes). In the future, we plan to apply these modifiers via [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver).
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/flex.md:34]:
```suggestion
Most of the flex layout variations are available at our standard [breakpoints](/developer-resources/global-css-variables). In the future, we plan to apply these modifiers via [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver). 
```

**Comment by mmenestr** [comment]
@edonehoo if this is ready to merge lmk! I approved it :)

**Comment by edonehoo** [comment]
@mmenestr it's ready to merge, ty!

**Pull request merged**